apps/datahub/src/app/organization/organization-details/organization-details.component.spec.ts

```diff
@@ -0,0 +1,317 @@
+        PreviousNextButtonsComponent,
+        UiElementsModule,
+        UiSearchModule,
+        MaxLinesComponent,
+        UiDatavizModule,
+        RouterLink,
+        UiWidgetsModule,
+        TranslateModule.forRoot(),
+        RouterTestingModule,
+      ],
+      providers: [
+        {
+          provide: OrganizationsServiceInterface,
+          useClass: OrganisationsServiceMock,
+        },
+        {
+          provide: SearchFacade,
+          useClass: SearchFacadeMock,
+        },
+        {
+          provide: ChangeDetectorRef,
+          useValue: changeDetectorRefMock,
+        },
+      ],
+    })
+      .overrideComponent(OrganizationDetailsComponent, {
+        set: {
+          changeDetection: ChangeDetectionStrategy.Default,
+        },
+      })
+      .compileComponents()
+
+    searchFacade = TestBed.inject(SearchFacade)
+
+    fixture = TestBed.createComponent(OrganizationDetailsComponent)
+    component = fixture.componentInstance
+    debugElement = fixture.debugElement
+
+    getHTMLElement = (dataTest: string) => {
+      const debugEl = debugElement.query(By.css(`[data-test="${dataTest}"]`))
+      return debugEl ? (debugEl.nativeElement as HTMLElement) : undefined
+    }
+
+    component.organization = anOrganizationWithManyDatasets
+
+    fixture.detectChanges()
+  })
+
+  it('should create', () => {
+    expect(component).toBeTruthy()
+  })
+
+  describe('Left column', () => {
+    describe('Organization description', () => {
+      it('should contain the organization description', () => {
+        const organizationDescriptionHtml = getHTMLElement(
+          'organizationDescription'
+        )
+
+        expect(organizationDescriptionHtml.textContent.trim()).toEqual(
+          anOrganizationWithManyDatasets.description?.trim()
+        )
+      })
+    })
+  })
+
+  describe('Right column', () => {
+    describe('Organization dataset count', () => {
+      it('should have the right count of dataset', () => {
+        const organizationDatasetCount = getHTMLElement(
+          'organizationDatasetCount'
+        ).querySelector('[data-test="figure"]')
+
+        expect(organizationDatasetCount.innerHTML).toEqual(
+          anOrganizationWithManyDatasets.recordCount?.toString()
+        )
+      })
+    })
+
+    describe('Organization email', () => {
+      it('should have the email button', () => {
+        const organizationEmail = getHTMLElement('organizationEmail')
+
+        expect(organizationEmail).toBeTruthy()
+
+        expect(organizationEmail?.getAttribute('href')).toEqual(
+          `mailto:${anOrganizationWithManyDatasets.email}`
+        )
+      })
+    })
+  })
+
+  describe('Last Published datasets', () => {
+    describe('Previous Next buttons', () => {
+      it('should not be displayed if organization is loading', () => {
+        organizationIsLoading.next(true)
+        fixture.detectChanges()
+
+        const organizationDetailsLastPublishedDatasetsPreviousNextButtons =
```

Comment:
Like many of the constants in this test files, this one is too long (and are hard to understand). Please rework their names :smile: